libs/sdk-ui-dashboard/src/model/commands/drill.ts

```diff
@@ -12,23 +12,42 @@ import { IInsight } from "@gooddata/sdk-model";
 import { IDashboardDrillEvent, IDrillDownDefinition } from "../../types";
 
 /**
- * Performs drill.
- *
  * @internal
  */
 export interface Drill extends IDashboardCommand {
     readonly type: "GDC.DASH/CMD.DRILL";
     readonly payload: {
+        /**
+         * Original drill event, that triggered this particular drill interaction.
+         */
         readonly drillEvent: IDashboardDrillEvent;
+        /**
+         * Context in which the drill interaction was triggered (widget and insight details - if available).
+         */
         readonly drillContext: DashboardDrillContext;
     };
 }
 
 /**
- * Creates the PerformDrill command.
- * Dispatching this command will result into triggering relevant drill events.
+ * Creates the {@link Drill} command.
+ * Dispatching this command will result into dispatching {@link DashboardDrillTriggered} event.
+ *
+ * This is general dashboard drill command with details about all possible more granular drill interactions that can follow.
+ * Reason for this general drill command is that it may happen that multiple drill interactions are possible for one drill event.
+ *
+ * Example: some attribute on the insight has drill down set and also widget has drill to insight set. Then this command must be dispatched with both
+ * {@link @gooddata/sdk-ui-ext#IDrillDownDefinition} and {@link @gooddata/sdk-backend-spi#IDrillToInsight} definitions.
+ *
+ * - This must be always the first command that occurs after the drill interaction and must be dispatched before more granular drill commands.
+ * - Specific drill commands that can follow this general drill command are: {@link DrillDown}, {@link DrillToInsight}, {@link DrillToDashboard},
+ *   {@link DrillToCustomUrl}, {@link DrillToAttributeUrl}, {@link DrillToLegacyDashboard}
+ *
  *
  * @internal
+ * @param drillEvent - original drill event, that triggered this particular drill interaction.
+ * @param drillContext - context in which the drill interaction was triggered (widget and insight details - if available).
+ * @param correlationId - optionally specify correlation id. It will be included in all events that will be emitted during the command processing.
+ * @returns dashboard command {@link Drill}.
```

Comment:
Nitpick: for return types, the link to the type will be generated automaticaly base don the return type so it is not necessary to specify it here as well (same for the paramters).

Reply:
This comment is applicable to other places in this file as well.